pkg/dockerregistry/server/blobdescriptorservice.go

```diff
@@ -65,17 +68,27 @@ func (bs *blobDescriptorService) Stat(ctx context.Context, dgst digest.Digest) (
 
 	context.GetLogger(ctx).Debugf("could not stat layer link %q in repository %q: %v", dgst.String(), repo.Named().Name(), err)
 
-	// verify the blob is stored locally
-	desc, err = dockerRegistry.BlobStatter().Stat(ctx, dgst)
-	if err != nil {
-		return desc, err
+	// only non-empty layers is wise to check for existence in the image stream.
+	// schema v2 has no empty layers.
+	if dgst.String() != digestSHA256GzippedEmptyTar.String() {
+		// ensure it's referenced inside of corresponding image stream
+		if !imageStreamHasBlob(repo, dgst) {
+			return distribution.Descriptor{}, distribution.ErrBlobUnknown
```

Comment:
@miminar It means that `imageStreamHasBlob` should be called only for local blobs ?
It seems you have me confused.